resolver/projects.go

```diff
@@ -40,14 +42,18 @@ func LocateGoModProjects(projects []GoModProject) (map[string]Dependency, error)
 }
 
 func LocateNpmPackageLockV3Projects(root string, projects []Project) (map[string]Dependency, error) {
-	nodeModulePrefixRegexp := regexp.MustCompile(`.*node_modules\/`)
 	deps := make(map[string]Dependency, len(projects))
 
 	for _, project := range projects {
 		baseDependencyName := nodeModulePrefixRegexp.ReplaceAllString(project.Name(), "")
```

Comment:
confirming: so for the example you gave, `node_modules/yup/node_modules/type-fest` turns into `type-fest` (see [here](https://go.dev/play/p/5LxIDZ6BbRP)). Is that expected?

Reply:
Yes that is expected, the goal is to get the name of the dependency, regardless of how nested it is in a parent `node_modules` hierarchy.